atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/service/SimpleTransactionService.java

```diff
@@ -52,12 +53,17 @@ public static SimpleTransactionService createV2(KeyValueService kvs) {
         return createSimple(kvs, TransactionConstants.TRANSACTIONS2_TABLE, TicketsEncodingStrategy.INSTANCE);
     }
 
-    public static SimpleTransactionService createV3(KeyValueService kvs, TaggedMetricRegistry metricRegistry) {
+    public static SimpleTransactionService createV3(
+            KeyValueService kvs, TaggedMetricRegistry metricRegistry, Supplier<Boolean> acceptStagingReadsAsCommitted) {
         if (kvs.getCheckAndSetCompatibility().consistentOnFailure()) {
             return createSimple(kvs, TransactionConstants.TRANSACTIONS2_TABLE, TicketsEncodingStrategy.INSTANCE);
```

Comment:
why does `createSimple` not take this flag? Is `createSimple` only used for tests?

Reply:
Would this even be safe for createSimple?